wagtail/users/views/users.py

```diff
@@ -180,57 +182,72 @@ def create(request):
     )
 
 
-@permission_required(change_user_perm)
-def edit(request, user_id):
-    user = get_object_or_404(User, pk=user_id)
-    can_delete = user_can_delete_user(request.user, user)
-    editing_self = request.user == user
+class Edit(EditView):
+
+    model = User
+    permission_policy = ModelPermissionPolicy(User)
+    form_class = get_user_edit_form()
+    template_name = "wagtailusers/users/edit.html"
+    index_url_name = "wagtailusers_users:index"
+    success_url = "wagtailusers_users:index"
```

Comment:
Pretty sure this is not needed, this is what Django CBVs use under the hood but we do not use this in our base class and instead use the `index_url_name` only.

```suggestion
```